src/bank.rs

```diff
@@ -942,23 +969,79 @@ impl Bank {
     pub fn set_finality(&self, finality: usize) {
         self.finality_time.store(finality, Ordering::Relaxed);
     }
+
+    pub fn store_account_subscription(
+        &self,
+        bank_sub_id: Pubkey,
+        pubkey: Pubkey,
+        sink: Sink<Account>,
+    ) {
+        let mut subscriptions = self.account_subscriptions.write().unwrap();
+        subscriptions.insert(bank_sub_id, (pubkey, sink));
+    }
+
+    pub fn remove_account_subscription(&self, bank_sub_id: &Pubkey) -> bool {
+        let mut subscriptions = self.account_subscriptions.write().unwrap();
+        subscriptions.remove(bank_sub_id).is_some()
+    }
+
+    fn check_account_subscriptions(&self, pubkey: &Pubkey, pre_userdata: &[u8], account: &Account) {
+        for (_bank_sub_id, (id, sink)) in self.account_subscriptions.read().unwrap().iter() {
```

Comment:
Little sucky that this will be `O(N)` with the number of active subscriptions (`check_signature_subscriptions` has the same runtime looks like).  

I'm ok with this for now but please file a follow-up performance issue for this so we can remember to maybe revisit later.  We can roll in potentially optimizing the `a.userdata.clone()`s in this issue as well.